webapp/channels/src/components/admin_console/system_properties/user_properties_values.tsx

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {FocusEventHandler, KeyboardEventHandler} from 'react';
+import React, {useMemo} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import type {GroupBase} from 'react-select';
+import {components} from 'react-select';
+import type {CreatableProps} from 'react-select/creatable';
+import CreatableSelect from 'react-select/creatable';
+
+import type {PropertyFieldOption, UserPropertyField} from '@mattermost/types/properties';
+
+import Constants from 'utils/constants';
+
+import {DangerText} from './controls';
+
+// import './user_properties_dot_menu.scss';
+
+type Props = {
+    field: UserPropertyField;
+    updateField: (field: UserPropertyField) => void;
+}
+
+type Option = {label: string; id: string; value: string};
+type SelectProps = CreatableProps<Option, true, GroupBase<Option>>;
+
+const UserPropertyValues = ({
+    field,
+    updateField,
+}: Props) => {
+    const {formatMessage} = useIntl();
+
+    const [query, setQuery] = React.useState('');
+    const isQueryValid = useMemo(() => !checkForDuplicates(field.attrs.options, query.trim()), [field?.attrs?.options, query]);
+
+    const addOption = (name: string) => {
+        const option: PropertyFieldOption = {
+            id: '',
+            name: name.trim(),
+        };
+
+        updateField({...field, attrs: {...field.attrs, options: [...field.attrs.options ?? [], option]}});
+    };
+
+    const setFieldOptions = (options: PropertyFieldOption[]) => {
+        updateField({...field, attrs: {...field.attrs, options}});
+    };
+
+    const processQuery = (query: string) => {
+        addOption(query);
+        setQuery('');
+    };
+
+    const handleKeyDown: KeyboardEventHandler = (event) => {
+        if (!query || !isQueryValid) {
+            return;
+        }
+
+        switch (event.key) {
```

Comment:
Should we use https://github.com/mattermost/mattermost/blob/b1a8810bd8beb2df73094be01f0e7926517543cc/webapp/channels/src/utils/constants.tsx#L1832 there?